core/src/main/scala/kafka/server/LegacyBroker.scala

```diff
@@ -48,10 +80,735 @@ object LegacyBroker {
+
+
+  private var logContext: LogContext = null
+
+  var kafkaYammerMetrics: KafkaYammerMetrics = null
+  var metrics: Metrics = null
+
+  val brokerState: BrokerState = new BrokerState
+
+  var dataPlaneRequestProcessor: KafkaApis = null
+  var controlPlaneRequestProcessor: KafkaApis = null
+
+  var authorizer: Option[Authorizer] = None
+  var socketServer: SocketServer = null
+  var dataPlaneRequestHandlerPool: KafkaRequestHandlerPool = null
+  var controlPlaneRequestHandlerPool: KafkaRequestHandlerPool = null
+
+  var logDirFailureChannel: LogDirFailureChannel = null
+  var logManager: LogManager = null
+
+  var replicaManager: ReplicaManager = null
+  var adminManager: LegacyAdminManager = null
+  var tokenManager: DelegationTokenManager = null
+
+  var dynamicConfigHandlers: Map[String, ConfigHandler] = null
+  var dynamicConfigManager: LegacyDynamicConfigManager = null
+  var credentialProvider: CredentialProvider = null
+  var tokenCache: DelegationTokenCache = null
+
+  var groupCoordinator: GroupCoordinator = null
+
+  var transactionCoordinator: TransactionCoordinator = null
+
+  var kafkaController: KafkaController = null
+
+  var brokerMetadataListener: BrokerMetadataListener = null
+
+  var brokerToControllerChannelManager: BrokerToControllerChannelManager = null
+
+  var kafkaScheduler: KafkaScheduler = null
+
+  var metadataCache: MetadataCache = null
+  var quotaManagers: QuotaFactory.QuotaManagers = null
+
+  val zkClientConfig: ZKClientConfig = LegacyBroker.zkClientConfigFromKafkaConfig(config).getOrElse(new ZKClientConfig())
+  private var _zkClient: KafkaZkClient = null
+
+  val correlationId: AtomicInteger = new AtomicInteger(0)
+  val brokerMetaPropsFile = "meta.properties"
+  val brokerMetadataCheckpoints = config.logDirs.map(logDir => (logDir, new BrokerMetadataCheckpoint(new File(logDir + File.separator + brokerMetaPropsFile)))).toMap
+
+  private var _clusterId: String = null
+  private var _brokerTopicStats: BrokerTopicStats = null
+
+  private var _featureChangeListener: FinalizedFeatureChangeListener = null
+
+  val brokerFeatures: BrokerFeatures = BrokerFeatures.createDefault()
+
+  val featureCache: FinalizedFeatureCache = new FinalizedFeatureCache(brokerFeatures)
+
+  def clusterId: String = _clusterId
+
+  // Visible for testing
+  private[kafka] def zkClient = _zkClient
+
+  private[kafka] def brokerTopicStats = _brokerTopicStats
+
+  private[kafka] def featureChangeListener = _featureChangeListener
+
+  newGauge("BrokerState", () => brokerState.currentState)
+  newGauge("ClusterId", () => clusterId)
+  newGauge("yammer-metrics-count", () =>  KafkaYammerMetrics.defaultRegistry.allMetrics.size)
+
+  val linuxIoMetricsCollector = new LinuxIoMetricsCollector("/proc", time, logger.underlying)
+
+  if (linuxIoMetricsCollector.usable()) {
+    newGauge("linux-disk-read-bytes", () => linuxIoMetricsCollector.readBytes())
+    newGauge("linux-disk-write-bytes", () => linuxIoMetricsCollector.writeBytes())
+  }
```

Comment:
kafka.metrics.MetricsTest is failing because a metric MBeanName changed again:  `kafka.server:type=LegacyBroker,name=ClusterId`